server/src/main/java/org/apache/iotdb/db/utils/FileLoaderUtils.java

```diff
@@ -64,4 +72,40 @@ public static void updateTsFileResource(TsFileMetaData metaData, TsFileSequenceR
       }
     }
   }
+
+  public static List<ChunkMetaData> loadChunkMetadataFromTsFileResource(
+      TsFileResource resource, Path seriesPath, QueryContext context) throws IOException {
+    List<ChunkMetaData> currentChunkMetaDataList;
+    if (resource == null) {
+      return new ArrayList<>();
+    }
+    if (resource.isClosed()) {
+      currentChunkMetaDataList = DeviceMetaDataCache.getInstance().get(resource, seriesPath);
+    } else {
+      currentChunkMetaDataList = resource.getChunkMetaDataList();
+    }
+    List<Modification> pathModifications =
+        context.getPathModifications(resource.getModFile(), seriesPath.getFullPath());
+
+    if (!pathModifications.isEmpty()) {
+      QueryUtils.modifyChunkMetaData(currentChunkMetaDataList, pathModifications);
+    }
+
+    for (ChunkMetaData data : currentChunkMetaDataList) {
+      if (data.getChunkLoader() == null) {
+        TsFileSequenceReader tsFileSequenceReader =
+            FileReaderManager.getInstance().get(resource, resource.isClosed());
+        data.setChunkLoader(new DiskChunkLoader(tsFileSequenceReader));
+      }
```

Comment:
Remove the if judgement, it will cause the file closed null pointer exception.
like:
for (ChunkMetaData data : currentChunkMetaDataList) {
TsFileSequenceReader tsFileSequenceReader =
             FileReaderManager.getInstance().get(resource, resource.isClosed());
         data.setChunkLoader(new DiskChunkLoader(tsFileSequenceReader));
}

it's ok to do so. because tsFileSequenceReader has been cached.

Reply:
removed